pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/filter/UserAuthCheckFilter.java

```diff
@@ -64,6 +68,35 @@ public void handle(Request request, Response response) {
             return;
         }
 
+        boolean canAccess = canAccess(request, ddpAuth);
+        int retries = 3;
+        if (!canAccess) {
+            do {
+                try {
+                    TimeUnit.SECONDS.sleep(1);
+                    log.warn("Retrying checking for access with non-cached permission data");
+                    try {
+                        canAccess = canAccess(request, RouteUtil.computeDDPAuth(request));
+                    } catch (TokenExpiredException e) {
+                        log.error("Found expired token for request", e);
+                        halt(401);
+                    } catch (Exception e) {
+                        log.error("Error while converting token for request", e);
+                        halt(401);
+                    }
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            } while (!canAccess && --retries > 0);
+        }
+
```

Comment:
This is really the meat of it. Filter first tries to check access using the previous existing code that can use a cached user info. If access is denied, then we are going to get the user info (DDPAuth) using the computedDDPAuth . This method avoids using the cached information.

Note here that an assumption is that the VAST majority of requests will be approved. And approved in the first pass, so this new code should not execute very often.

Reply:
Why do you need 3 attempts?

Reply:
You know what. I think this is probably not necessary. I was worrying about request scenarios that are not realistic.
I am going to just keep it to 1 try and test it out.

Reply:
Removed 902feb3fcdb4a31434dd77a66356c08113a6564f